src/matrix_ops.hpp

```diff
@@ -34,4 +33,38 @@ requires(matrix<V, Cols>::is_square)
     (std::make_index_sequence<Cols>{});
 }
 
+template <Vector V, std::size_t Cols, Vector W, std::size_t ColsOther>
+requires(std::is_same<typename extend_to<V, 1>::type,
+                      typename extend_to<W, 1>::type>::value&&
+             std::is_same<scalar_t<V>, scalar_t<W>>::
+                 value) constexpr auto diagonal(matrix<V, Cols> const& m0,
+                                                matrix<W, ColsOther> const& m1)
+{
+    using U = typename extend_to<V, V::size + W::size>::type;
+    matrix<U, Cols + ColsOther> diagm{};
+    std::transform(
+        m0.begin(),
+        m0.end(),
+        diagm.begin(),
```

Comment:
Shouldn't the first 3 be const qualified? Why is `diagm` repeated?

Reply:
Mm, I did not want to create a zero vector with the composition dimensions (Rows+RowsOther), copy the first Row  into it, and then copy it back into the column of diagm - I thought I could just work on the column of diagm.

Reply:
Let's address this later